### PR TITLE
[FLINK-32741]Remove DataSet related descriptions in doc

### PR DESCRIPTION
**Pull request by pegasas**
## What is the purpose of the change

Since All DataSet APIs will be deprecated in [FLINK-32558](https://issues.apache.org/jira/browse/FLINK-32558) and we don't recommend developers to use the DataSet, the descriptions of DataSet should be removed in the doc after [FLINK-32558](https://issues.apache.org/jira/browse/FLINK-32558).

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluster with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by pegasas** [comment]
Hi, @xintongsong, @WencongLiu ,

I deprecated all files on /dev/dataset on content/ and content.zh
in my private view, all dataset sub-document should be deprecated now. 
 https://github.com/apache/flink/blob/master/docs/content.zh/docs/dev/dataset/hadoop_compatibility.md for this I am not definitely sure about that, but in my user experience, It seems we have FLIP-27/143 FileSource/FileSink which could directly write to local/distributed file system? not sure if it is outdated.

And I correct several files as a result and I search 'Dataset' as a keyword. I correct them with my limited understanding and user experience.

feel free to correct me if there is any concern.

**Comment by xintongsong** [comment]
Thanks, @pegasas.

I think https://github.com/apache/flink/blob/master/docs/content.zh/docs/dev/dataset/hadoop_compatibility.md should be removed. Actually, this file only exist for the `content.zh` but not for `content`. If you check the git history, you'll find that hadoop format related documents have been moved to `connectors/datastream/formats` in FLINK-21407, where the Chinese translation was overlooked.

I'll review the other changes asap.

File: docs/content/docs/concepts/stateful-stream-processing.md [hunk 1]
```diff
@@ -342,26 +342,3 @@ give *exactly once* guarantees even in *at least once* mode.
...
-    streams.  That is possible, because inputs are bounded. This pushes the
-    cost more towards the recovery, but makes the regular processing cheaper,
-    because it avoids checkpoints.
-
-  - Stateful operations in the DataSet API use simplified in-memory/out-of-core
-    data structures, rather than key/value indexes.
-
-  - The DataSet API introduces special synchronized (superstep-based)
-    iterations, which are only possible on bounded streams. For details, check
-    out the [iteration docs]({{< ref "docs/dev/dataset/iterations" >}}).
```
Comment (xintongsong) [inline docs/content/docs/concepts/stateful-stream-processing.md:365]: I don't think these should all be removed. These contents also applies to DataStream jobs in batch execution mode.

I'd suggest:
1. Link `batch programs` to DataStream Batch Execution mode: `docs/dev/datastream/execution_mode`
2. Remove "A *DataSet* is treated internally as a stream of data."
3. Change "Stateful operations in the DataSet API" to "State backend in batch execution mode".
4. Remove "The DataSet API introduces special synchronized (superstep-based) iterations, which are only possible on bounded streams. For details, check out the [iteration docs]({{< ref "docs/dev/dataset/iterations" >}}).".

File: docs/content/docs/dev/datastream/scala_api_extensions.md [hunk 2]
```diff
@@ -198,14 +198,8 @@ To use this extension exclusively, you can add the following `import`:
...
 {{< /highlight >}}
 
-for the DataSet extensions and
-
-{{< highlight scala >}}
-import org.apache.flink.streaming.api.scala.extensions.acceptPartialFunctions
-{{< /highlight >}}
-
 The following snippet shows a minimal example of how to use these extension
-methods together (with the DataSet API):
```
Comment (xintongsong) [inline docs/content/docs/dev/datastream/scala_api_extensions.md:208]: It's probably out of the scope of this PR. I wonder if we should simply also remove all Scala API documentations.

File: docs/content/docs/ops/state/task_failure_recovery.md [hunk 3]
```diff
@@ -334,7 +334,7 @@ A region is a set of tasks that communicate via pipelined data exchanges.
 That is, batch data exchanges denote the boundaries of a region.
 - All data exchanges in a DataStream job or Streaming Table/SQL job are pipelined.
 - All data exchanges in a Batch Table/SQL job are batched by default.
-- The data exchange types in a DataSet job are determined by the 
+- The data exchange types in a DataStream job are determined by the
```
Comment (xintongsong) [inline docs/content/docs/ops/state/task_failure_recovery.md:337]: I think this page is outdated. Currently, all DataStream / Table / SQL jobs should rely on `ExecutionMode` for determining the data exchange type. We probably can combine these 3 items into 1.

**Review by xintongsong — COMMENTED** [review]
Thanks for the effort, @pegasas. I left some comments, PTAL.

In addition to the inline comments:
1. There are [breaking links](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=51971&view=logs&j=c5d67f7d-375d-5407-4743-f9d0c4436a81&t=38411795-40c9-51fa-10b0-bd083cf9f5a5&l=51) that cause CI failures. You can build the document locally to find all breaking links. See also `docs/README.md`.
2. I think `docs/content[.zh]/docs/connectors/dataset` should also be removed.

File: docs/content/docs/concepts/stateful-stream-processing.md (hunk 1 above)
Comment (pegasas) [reply docs/content/docs/concepts/stateful-stream-processing.md:365]: sure. 
This is a good point.
I will search the others to see if any other same pattern, which could be executed in BATCH mode.

File: docs/content/docs/dev/datastream/scala_api_extensions.md (hunk 2 above)
Comment (pegasas) [reply docs/content/docs/dev/datastream/scala_api_extensions.md]: Sure. I saw there is another work item [[FLIP-265] Remove all Scala APIs](https://issues.apache.org/jira/browse/FLINK-29741) in 2.0 burndown, maybe we can track in that thread.

File: docs/content/docs/ops/state/task_failure_recovery.md (hunk 3 above)
Comment (pegasas) [reply docs/content/docs/ops/state/task_failure_recovery.md]: Sure. Merged.

**Review by xintongsong — APPROVED** [review]
@pegasas, thanks for addressing the comments. LGTM. Merging.

**Comment by pegasas** [comment]
> @pegasas, thanks for addressing the comments. LGTM. Merging.

Thanks @xintongsong for reviewing and suggestions!

**Pull request closed without merging**